src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -513,6 +513,61 @@ void SQLStorage::clearMetadata() {
   }
 }
 
+void SQLStorage::storeDelegation(const std::string& data, const Uptane::Role role) {
+  SQLite3Guard db = dbConnection();
+
+  if (!db.beginTransaction()) {
+    LOG_ERROR << "Can't start transaction: " << db.errmsg();
+    return;
+  }
+
+  auto del_statement = db.prepareStatement<std::string>("DELETE FROM delegations WHERE role_name=?;", role.ToString());
```

Comment:
I think it could be replace by a single "INSERT OR REPLACE"

Reply:
Fair point. I copied the style of the `NonRoot` functions. Why don't those do the same?

Reply:
`meta` has a `UNIQUE(repo, meta_type, version)` but the delete is `DELETE FROM meta WHERE (repo=? AND meta_type=?);`.

With the current schema, `INSERT OR REPLACE` would allow multiple versions to accumulate while the current `storeNonRoot` only keep the last updated version. To be honest, I'm not sure which would be the preferred semantics. @OYTIS, any input?

Reply:
@lbonn Yes, the preferred semantics was to only keep the last version. Again `ORDER BY` and `LIMIT` are just for precaution, also known as defensive programming, which is not always considered a good style. We can delete them if you feel like it.

Reply:
I'm not necessarily against defensive programming. Thing is, in this case the schema doesn't protect against multiple version. So should the unique constraint be only `UNIQUE(repo, meta_type)`? I find it weird that we keep this constraint in code only.